File: AndroidBasics/notes/src/main/java/pan/alexander/notes/domain/entities/Note.java
package pan.alexander.notes.domain.entities;

import java.util.Objects;

public class Note {
    private String title;
    private String description;
    private long date;
    private String color;

    public Note(String title, String description, long date, String color) {
        this.title = title;
        this.description = description;
        this.date = date;
        this.color = color;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Note note = (Note) o;
        return date == note.date &&
                title.equals(note.title) &&
                description.equals(note.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(title, description, date);
    }
}

File: PictureoftheDay/app/src/main/java/pan/alexander/pictureoftheday/domain/notes/entities/Note.kt
package pan.alexander.pictureoftheday.domain.notes.entities

import android.content.Context
import androidx.annotation.Keep
import androidx.core.content.ContextCompat
import pan.alexander.pictureoftheday.R
import java.util.*

@Keep
data class Note(
    val id: Int,
    var text: String,
    var date: Date,
    var priority: Priority
) {

    constructor(
        text: String,
        date: Date,
        priority: Priority
    ) : this(
        globalIds++,
        text,
        date,
        priority
    )

    enum class Priority {
        LOW,
        NORMAL,
        HIGH;

        fun getColor(context: Context): Int {
            return when (this) {
                LOW -> ContextCompat.getColor(context, R.color.cyan_200)
                NORMAL -> ContextCompat.getColor(context, R.color.white)
                HIGH -> ContextCompat.getColor(context, R.color.pink_200)
            }
        }
    }

    private companion object {
        var globalIds: Int = 1
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Note

        if (id != other.id) return false
        if (text != other.text) return false
        if (date != other.date) return false
        if (priority != other.priority) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + text.hashCode()
        result = 31 * result + date.hashCode()
        result = 31 * result + priority.hashCode()
        return result
    }

}
